tests/codegen/align-fn.rs

```diff
@@ -7,3 +7,12 @@
 #[no_mangle]
 #[repr(align(16))]
 pub fn fn_align() {}
+
+pub struct A;
+
+impl A {
+    // CHECK: align 16
+    #[no_mangle]
+    #[repr(align(16))]
+    pub fn method_align(self) {}
+}
```

Comment:
Can you also add a check for trait methods and associated functions, just to be sure? i.e.
```suggestion

    // CHECK: align 16
    #[no_mangle]
    #[repr(align(16))]
    pub fn associated_fn() {}
}

trait T {
    fn trait_fn() {}
    fn trait_method(self) {}
}

impl T for A {
    // CHECK: align 16
    #[no_mangle]
    #[repr(align(16))]
    fn trait_fn() {}
    
    // CHECK: align 16
    #[no_mangle]
    #[repr(align(16))]
    fn trait_method(self) {}
}
```

Reply:
Should those be allowed? The current change doesn't allow those, but I can change it to include trait impls as well.

Reply:
I think yes, they should? Don't really see a reason to not allow these (btw maybe a test for default bodies should be added too, although I'm not sure if it's easy to do...).